### ISSUE TITLE
Minimum storage used by control curve parameters

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#368]
Some of the control curve parameters do an interpolation when a storage falls below it's control curve, like this:

```
bottom = 0.0
weight = (current percent - bottom) / (control curve - bottom)
```

Should we be using the minimum storage attribute for the value of `bottom` instead of the control curve? Two storages with the same amount of available water could report completely different resource states.

**Comment by snorfalorpagus** [comment]
We discussed this further today in the context of a storage that has a negative minimum volume.

I confirmed the behaviour of Aquator in this respect is to always calculate resource state from empty (i.e. 0 volume). The emergency and dead storage volumes stored on a reservoir do no impact it's resource state.

**Comment by jetuk** [comment]
I think we should do it on the difference between min and max. If not we should not support -ve volumes.

**Comment by jetuk** [comment]
This can be closed as won't fix? We're keeping zero for control curve interpolation and not supporting negative volumes?

**Comment by snorfalorpagus** [comment]
I agree we shouldn't support negative volumes, but I think we need an option to have control curves calculated from a non-empty value.
